packages/platform-sdk-support/src/bignumber.ts

```diff
@@ -264,26 +276,22 @@ export class BigNumber {
 		return this.#value.isLessThanOrEqualTo(this.toBigNumber(value));
 	}
 
-	/**
-	 * Creates an instance of BigNumber with the current value multiplied by one satoshi.
-	 *
-	 * @returns {BigNumber}
-	 * @memberof BigNumber
-	 */
-	public toSatoshi(): BigNumber {
-		return BigNumber.make(this.#value.multipliedBy(1e8));
-	}
-
 	/**
 	 * Divides the current value by one satoshi and rounds it to the given amount of decimals.
 	 *
 	 * @param {number} [decimals=8]
 	 * @returns {string}
 	 * @memberof BigNumber
 	 */
-	public toHuman(decimals = 8): string {
+	public toHuman(decimals?: number): string {
+		decimals = decimals || this.#decimals || 8;
+		if (decimals === undefined) {
+			// TODO: remove 8 above and uncomment below once everything has been ported
+			// throw new Error("Specify number of decimals for using toHuman()");
+		}
+		const denominator = this.powerOfTen(decimals);
 		return BigNumber.make(this.#value)
-			.divide(1e8)
+			.divide(denominator)
 			.toFixed(decimals);
 	}
```

Comment:
Does this function look right? I'm not sure because I struggle to write test cases that make sense

Reply:
That does look like it should be. What's the issue with writing a test for it?

Reply:
Idk everything I try return the wrong thing. Can you pls give 2-3 test examples using decimals both in make() and toHuman() similar to:
```typescript
	expect(BigNumber.make(123.456 * 1e8).toHuman()).toBe("123.45600000");
	expect(BigNumber.make(123.456789 * 1e8).toHuman()).toBe("123.45678900");
```

Reply:
```ts
test.only.each([
	[2, 1e2, "100.00"],
	[3, 1e3, "100.000"],
	[4, 1e4, "100.0000"],
	[5, 1e5, "100.00000"],
	[6, 1e6, "100.000000"],
	[7, 1e7, "100.0000000"],
	[8, 1e8, "100.00000000"],
	[9, 1e9, "100.000000000"],
	[10, 1e10, "100.0000000000"],
])("#toHuman(%s)", (decimals, multiplier, expected) => {
	expect(BigNumber.make(100 * multiplier, decimals).toHuman()).toBe(expected);
});
```

Something like this should probably result in the behaviour you want.

Reply:
Like this error:
![image](https://user-images.githubusercontent.com/1091900/120356206-5b101c00-c304-11eb-99cd-f4dfc4487c66.png)

Reply:
Ok I think I got it

Reply:
The issue with the screenshot is that you multiplied it by `1e8` instead of `1e6` so calling `toHuman(6)` will result in a larger value since there's a difference of `e2` left.